lib/portage/dbapi/bintree.py

```diff
@@ -504,31 +542,52 @@ def move_ent(self, mylist, repo_match=None):
 				writemsg("!!! "+mycpv+" -> "+mynewcpv+"\n", noiselevel=-1)
 				continue
 
-			tbz2path = self.getname(mycpv)
-			if os.path.exists(tbz2path) and not os.access(tbz2path,os.W_OK):
+			binpkg_path = self.getname(mycpv)
+			if os.path.exists(binpkg_path) and \
+				not os.access(binpkg_path,os.W_OK):
 				writemsg(_("!!! Cannot update readonly binary: %s\n") % mycpv,
 					noiselevel=-1)
 				continue
 
 			moves += 1
-			mytbz2 = portage.xpak.tbz2(tbz2path)
-			mydata = mytbz2.get_data()
+			if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+				mytbz2 = portage.xpak.tbz2(binpkg_path)
+				mydata = mytbz2.get_data()
+			elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+				mybinpkg = portage.gpkg.gpkg(self.settings, mycpv, binpkg_path)
+				mydata = mybinpkg.get_metadata()
+			else:
+				continue
+
 			updated_items = update_dbentries([mylist], mydata, parent=mycpv)
 			mydata.update(updated_items)
-			mydata[b'PF'] = \
-				_unicode_encode(mynewpkg + "\n",
-				encoding=_encodings['repo.content'])
-			mydata[b'CATEGORY'] = \
-				_unicode_encode(mynewcat + "\n",
-				encoding=_encodings['repo.content'])
+			if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
```

Comment:
Do my eyes deceive me or are both branches the same?

Reply:
The old xpak use binary as key, GPKG always use UTF-8. I haven't figured out whether compatibility should be done in gpkg or they should be separated in others code.